aten/src/ATen/xpu/XPUEvent.h

```diff
@@ -156,6 +163,24 @@ struct TORCH_XPU_API XPUEvent {
   }
 
  private:
+  void assignEvent(sycl::queue& queue) {
+#if SYCL_COMPILER_VERSION >= 20250000
+    if (enable_timing_) {
+      event_ = std::make_unique<sycl::event>(
+          sycl::ext::oneapi::experimental::submit_profiling_tag(queue));
+    } else {
+      event_ = std::make_unique<sycl::event>(queue.ext_oneapi_submit_barrier());
```

Comment:
We plan to use `ext_oneapi_get_last_event` to replace `ext_oneapi_submit_barrier` when recording an event without the `enable_timing` feature.
This code change will happen while CI uplifts to 2025.0.